### ISSUE TITLE
Dagit Execution Visualization Issues When Running Subset of DAG - 0.13.4

### ISSUE THREAD

**Issue by cwe5590** [dagster-io/dagster#5616]
## Summary

Running seemingly valid job causes issues in dagit. No execution visualizations are observable.

## Reproduction

```
Operation name: RunRootQuery

Message: '_TypeHintInferred[src.ops.data_pipelines.DBLoadMetaData]'

Path: ["pipelineRunOrError","executionPlan","steps",2,"inputs",0,"dependsOn",0,"outputs",0,"type"]

Locations: [{"line":51,"column":13},{"line":105,"column":11}]

Stack Trace:
  File "/Users/chrisevans/Repositories/data-platform/orchestrator/bi/.venv/lib/python3.9/site-packages/graphql/execution/executor.py", line 452, in resolve_or_error
    return executor.execute(resolve_fn, source, info, **args)
  File "/Users/chrisevans/Repositories/data-platform/orchestrator/bi/.venv/lib/python3.9/site-packages/graphql/execution/executors/sync.py", line 16, in execute
    return fn(*args, **kwargs)
  File "/Users/chrisevans/Repositories/data-platform/orchestrator/bi/.venv/lib/python3.9/site-packages/dagster_graphql/schema/execution.py", line 36, in resolve_type
    return to_dagster_type(self._pipeline_snapshot, self._step_output_snap.dagster_type_key)
  File "/Users/chrisevans/Repositories/data-platform/orchestrator/bi/.venv/lib/python3.9/site-packages/dagster_graphql/schema/dagster_types.py", line 25, in to_dagster_type
    dagster_type_meta = pipeline_snapshot.dagster_type_namespace_snapshot.get_dagster_type_snap(
  File "/Users/chrisevans/Repositories/data-platform/orchestrator/bi/.venv/lib/python3.9/site-packages/dagster/core/snap/dagster_types.py", line 48, in get_dagster_type_snap
    return self.all_dagster_type_snaps_by_key[key]
```

## Dagit UI/UX Issue Screenshots

![image](https://user-images.githubusercontent.com/45503169/141218693-6616c46c-03d2-4404-861f-140e5ba89e0c.png)

## Additional Info about Your Environment

Experience locally in 0.13.4

---
#### Message from the maintainers:
Impacted by this bug? Give it a 👍. We factor engagement into prioritization.

**Comment by hellendag** [comment]
This doesn't look like a Dagit-specific issue. Practitioner team, do you guys know what might lead to this error?

**Comment by sryza** [comment]
@yuhan might know

**Comment by yuhan** [comment]
Hi @cwe5590, thanks for reporting this issue. To better investigate the error, I'm going to ask for some additional details:
- Was the retry a re-execute all or a subset re-execution?
- Was the original run a full-graph execution? in other words, did you specify any op selection?
We've recently found a bug related to run snapshot inconsistent with op selection which caused a framework error in the same stack trace, and a fix went in the `0.13.8` release. Sorry for the thrash - Do you mind upgrade to 0.13.8 and see if launching a brand new run and re-executing from there still hit this error?

**Comment by yuhan** [comment]
Closing as it should have been fixed in 0.13.8. Please reopen if there's still an issue.
